connectors/sources/jira.py

```diff
@@ -320,8 +371,310 @@ def get_default_configuration(cls):
+            )
+
+        return document
+
+    async def _user_information_list(self, project):
+        async for response in self.jira_client.api_call(
+            url_name=PROJECT_PERMISSIONS, project_key=project["key"]
+        ):
+            yield await response.json()
+
+    async def _project_access_control(self, project):
+        if not self._dls_enabled():
+            return []
+
+        access_control = set()
+        async for actors in self._user_information_list(project=project):
+            for actor in actors:
+                if actor["accountType"] == ATLASSIAN:
+                    access_control.add(
+                        _prefix_account_id(user_id=actor.get("accountId"))
+                    )
+                    access_control.add(_prefix_username(user=actor.get("displayName")))
+        return list(access_control)
+
+    async def _issue_security_level(self, issue_key):
+        async for response in self.jira_client.api_call(
+            url_name=ISSUE_SECURITY_LEVEL, issue_key=issue_key
+        ):
+            yield await response.json()
+
+    async def _issue_security_level_members(self, level_id):
+        async for response in self.jira_client.paginated_api_call(
+            url_name=SECURITY_LEVEL_MEMBERS, level_id=level_id
+        ):
+            yield response
+
+    async def _project_role_members(self, project, role_id, access_control):
+        async for actor_response in self.jira_client.api_call(
+            url_name=PROJECT_ROLE_MEMBERS,
+            project_key=project["key"],
+            role_id=role_id,
+        ):
+            actors = await actor_response.json()
+            for actor in actors.get("actors", []):
+                if actor.get("actorUser"):
+                    access_control.add(
+                        _prefix_account_id(
+                            user_id=actor.get("actorUser").get("accountId")
+                        )
+                    )
+                    access_control.add(_prefix_username(user=actor.get("displayName")))
+                elif actor.get("actorGroup"):
+                    access_control.add(
+                        _prefix_group_id(
+                            group_id=actor.get("actorGroup").get("groupId")
+                        )
+                    )
+            yield access_control
+
+    async def _issue_access_control(self, issue_key, project):
+        if not self._dls_enabled():
+            return []
+
+        access_control = set()
+        async for response in self._issue_security_level(issue_key=issue_key):
+            if security := response.get("fields", {}).get("security"):
+                level_id = security.get("id")
+                async for members in self._issue_security_level_members(
+                    level_id=level_id
+                ):
+                    for actor in members["values"]:
+                        if actor.get("holder", {}).get("type") == "user":
```

Comment:
suggestion: `actor.get("holder", {}).get("type")` is used at three different places. I would suggest to use a variable here.